pmd-core/src/main/java/net/sourceforge/pmd/lang/ast/QualifiedName.java

```diff
@@ -0,0 +1,36 @@
+/**
+ * BSD-style license; for more info see http://pmd.sourceforge.net/license.html
+ */
+
+package net.sourceforge.pmd.lang.ast;
+
+/**
+ * Basic interface for qualified names. That's used in the metrics framework to refer unambiguously to operations or
+ * classes. Language specific pmd modules should have at most one implementation of this interface, to allow safe
+ * downcasting from QualifiedName to e.g. JavaQualifiedName.
```

Comment:
Just a comment: When would it be necessary to downcast to the concrete implementation? I guess, it's ok inside the implementation of a specific language module (since it's implementation specific) - and this rule makes life easier.